src/Ae.js

```diff
@@ -84,33 +84,32 @@ export default class Ae {
     rawTxHex: string,
     networkId: string = "ae_mainnet"
   ): Promise<string> {
+    const [major, minor] = (await this._getAppConfiguration()).version.split(".").map(n => +n);
+    const acceptsTransactionLength = !(major === 0 && minor < 4);
     let offset = 0;
+    const headerLength  = 4 + 1 + (acceptsTransactionLength ? 4 : 0);
```

Comment:
We can naturally calculate this length by writing the header (that we have to do in the first chunk). So, proposing to write header firstly and then to take it length.